src/commands/deploy/genezio.ts

```diff
@@ -234,6 +234,30 @@ export async function genezioDeploy(options: GenezioDeployOptions) {
                 await doAdaptiveLogAction(
                     `Running frontend ${index + 1} deploy script`,
                     async () => {
+                        const environment = frontend.environment;
+                        if (environment) {
+                            const newEnvObject: Record<string, string> = {};
+
+                            await Promise.all(
+                                Object.keys(environment).map(async (key) => {
+                                    const resolvedValue = await evaluateResource(
+                                        configuration,
+                                        environment[key],
+                                        options.stage,
+                                        /* envFile */ undefined,
+                                    );
+                                    newEnvObject[key] = resolvedValue;
+                                }),
```

Comment:
I'm pretty sure this won't work as expected as the map function does not return a Promise. Promise.all takes an array of promises and runs them in parallel.